conans/client/conf/detect.py

```diff
@@ -248,6 +264,9 @@ def _detect_os_arch(result, output):
     the_os = detected_os()
     result.append(("os", the_os))
     result.append(("os_build", the_os))
+    the_subsystem = detect_subsystem()
+    if the_subsystem:
+        result.append(("os.subsystem", the_subsystem))
```

Comment:
This subsetting was invented to distinguish the "native" built apps for the subsystem (using internal dll) or native windows apps. 
That you are running Conan in a subsystem it doesn't mean that you are building subsystem-native apps, for example, you can run MSYS2 in mingw mode to build windows native apps.

Reply:
so, remove it?

Reply:
I would say, yes, remove it. The subsystem subsetting shouldn't be autodetected.

Reply:
sure, will do

Reply:
@lasote done